diff --git a/frameworks/js/camera_napi/src/session/camera_session_napi.cpp b/frameworks/js/camera_napi/src/session/camera_session_napi.cpp
index 6db12850..e89cb22b 100644
--- a/frameworks/js/camera_napi/src/session/camera_session_napi.cpp
+++ b/frameworks/js/camera_napi/src/session/camera_session_napi.cpp
@@ -14,6 +14,7 @@
  */
 
 #include "session/camera_session_napi.h"
+#include "output/photo_output_napi.h"
 #include <uv.h>
 
 namespace OHOS {
diff --git a/interfaces/kits/js/camera_napi/include/session/camera_session_napi.h b/interfaces/kits/js/camera_napi/include/session/camera_session_napi.h
index 6a7d75e8..acf46971 100644
--- a/interfaces/kits/js/camera_napi/include/session/camera_session_napi.h
+++ b/interfaces/kits/js/camera_napi/include/session/camera_session_napi.h
@@ -25,7 +25,6 @@
 #include "session/capture_session.h"
 
 #include "input/camera_input_napi.h"
-#include "output/photo_output_napi.h"
 #include "output/preview_output_napi.h"
 #include "output/video_output_napi.h"
 #include "output/metadata_output_napi.h"
